rusk-wallet/src/lib/gql.rs

```diff
@@ -0,0 +1,150 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use gql_client::Client;
+use serde::Deserialize;
+use serde_json::Value;
+use tokio::runtime::Handle;
+use tokio::task::block_in_place;
+
+use super::error::GraphQLError;
+
+/// GraphQL is a helper struct that aggregates all queries done
+/// to the Dusk GraphQL database.
+/// This helps avoid having helper structs and boilerplate code
+/// mixed with the wallet logic
+#[derive(Clone, Debug)]
+pub struct GraphQL {
+    url: String,
+}
+
+/// Transaction status
+#[derive(Debug)]
+pub enum TxStatus {
+    Ok,
+    NotFound,
+    Error(String),
+}
+
+impl GraphQL {
+    /// Create a new GraphQL wallet client
+    pub fn new(url: &str) -> Self {
+        Self {
+            url: url.to_string(),
+        }
```

Comment:
If we perform implicit conversion, we should either take argument as generic, or as the concrete final type (`String`).

Example:

```suggestion
    pub fn new<S>(url: S) -> Self where S: Into<String> {
        Self {
            url: url.into(),
        }
```